itest/lnd_misc_test.go

```diff
@@ -1242,3 +1243,47 @@ func testSignVerifyMessageWithAddr(ht *lntest.HarnessTest) {
 
 	require.False(ht, respValid.Valid, "external signature did validate")
 }
+
+// testInvoiceCltvExpiryFlag ensures that the CltvExpiry option passed to the
+// AddInvoice RPC interface sets `min_final_cltv_expiry_delta` in the invoice
+// and that it fails when a cltv too small or too large is passed.
+func testInvoiceCltvExpiryFlag(ht *lntest.HarnessTest) {
+	// Set up a node
+	alice := ht.Alice
+
+	// Create an invoice with a CltvExpiry set
+	const paymentAmt = 1000
+	minCltvDelta := uint64(100)
+	invoice := &lnrpc.Invoice{
+		Value:      paymentAmt,
+		CltvExpiry: minCltvDelta,
+	}
+
+	// Add the invoice and use the decodepayreq rpc to parse it
+	invoiceResp := alice.RPC.AddInvoice(invoice)
+	payReq := alice.RPC.DecodePayReq(invoiceResp.PaymentRequest)
+
+	// Requested CltvExpiry must equal what decodepayreq reports as the
+	// cltv_expiry of the invoice
+	require.Equal(ht, uint64(payReq.CltvExpiry), minCltvDelta)
+
+	// Set the min_cltv delta to a value lower than router.MinCLTVDelta
+	// which is the minimum value permitted. (18 in February 2024) and
+	// assert an error when adding invoice
+	minCltvDelta = uint64(17)
+	invoice = &lnrpc.Invoice{
+		Value:      paymentAmt,
+		CltvExpiry: minCltvDelta,
+	}
+	alice.RPC.AddInvoiceAssertErr(invoice)
```

Comment:
i think it would be nice if we could assert the specific error we are getting using "require.ErrorContains"

Reply:
Done